commit dc0b877bf55f4d3a969015101d3bf9e738c4f0a9
Author: popcornmix <popcornmix@gmail.com>
Date:   Tue May 8 23:14:05 2012 +0100

    Fix BUG() in bcm2708_fb_set_par doesn't work if the console is currently locked. Thanks lp0

Index: linux-3.2.46/drivers/video/bcm2708_fb.c
===================================================================
--- linux-3.2.46.orig/drivers/video/bcm2708_fb.c	2013-07-26 19:32:52.000000000 +0000
+++ linux-3.2.46/drivers/video/bcm2708_fb.c	2013-07-26 19:32:58.000000000 +0000
@@ -27,6 +27,7 @@
 #include <linux/platform_device.h>
 #include <linux/clk.h>
 #include <linux/printk.h>
+#include <linux/console.h>
 
 #include <mach/platform.h>
 #include <mach/vcio.h>
@@ -243,8 +244,13 @@
 			iounmap(fb->fb.screen_base);
 		fb->fb.screen_base =
 			(void *)ioremap_wc(fb->fb.fix.smem_start, fb->fb.screen_size);
-		if (!fb->fb.screen_base)
+		if (!fb->fb.screen_base) {
+			/* the console may currently be locked */
+			console_trylock();
+			console_unlock();
+
 			BUG();		/* what can we do here */
+		}
 	}
 	pr_info
 	    ("BCM2708FB: start = %p,%p width=%d, height=%d, bpp=%d, pitch=%d size=%d success=%d\n",
